app/connector/SchemeConnector.scala

```diff
@@ -192,4 +144,59 @@ class SchemeConnectorImpl @Inject()(
           invalidPayloadHandler.logFailures("/resources/schemas/schemeVariationSchema.json", data)
       }
   }
+
+  private def desHeader(implicit hc: HeaderCarrier): Seq[(String, String)] = {
+    val requestId = getCorrelationId(hc.requestId.map(_.value))
+
+    Seq("Environment" -> config.desEnvironment, "Authorization" -> config.authorization,
+      "Content-Type" -> "application/json", "CorrelationId" -> requestId)
+  }
+
+  private def handleSchemeDetailsResponse(psaId: String, response: HttpResponse, url: String)(
+    implicit requestHeader: RequestHeader, executionContext: ExecutionContext) = {
+
+    val badResponseSeq = Seq("INVALID_CORRELATION_ID", "INVALID_PAYLOAD", "INVALID_IDTYPE", "INVALID_SRN", "INVALID_PSTR", "INVALID_CORRELATIONID")
+
+    response.status match {
+      case OK =>
+        val temporaryMappingTest = response.json.transform(schemeSubscriptionDetailsTransformer.transformToUserAnswers)
+
+        if (temporaryMappingTest.isSuccess)
```

Comment:
Move this to line 177 - it's unnecessary to do this if variations switched off.

Reply:
added to get the confidence before hand for conversion